torch/distributed/tensor/_dtensor_spec.py

```diff
@@ -308,10 +312,12 @@ def _maybe_convert_StridedShard_to_shard_order(
                 )
         return tuple(shard_order)
 
-    def _verify_shard_order(self, shard_order: ShardOrder) -> None:
+    def _verify_shard_order(
```

Comment:
unclear who should call this and why, and why its possible for shard_order to be incorrect in the first place, 

also, like the other staticmethods, seems more like a util in placements module than a staticmethod on spec.